examples/cc3200/xively_firmware_updates/main.c

```diff
@@ -0,0 +1,1249 @@
+
+/* Xively Functions */
+void ConnectToXively();
+void on_connected( xi_context_handle_t in_context_handle, void* data, xi_state_t state );
+void send_fileinfo( const xi_context_handle_t context_handle,
+                    const xi_timed_task_handle_t timed_task_handle,
+                    void* user_data );
+void pushButtonInterruptHandlerSW2();
+void pushButtonInterruptHandlerSW3();
+
+/* Xively and Wifi Config File Parsing functions */
+static void parseCredentialsFromConfigFile();
+static int8_t parseKeyValue( config_entry_t* config_file_context,
+                             const char* key,
+                             const uint8_t is_required,
+                             char** out_value );
+static int8_t mapWifiSecurityTypeStringToInt( const char* security_type );
+
+/* Xively and Wifi Config value verification functions */
+static int8_t validateWifiConfigurationVariables( const char* wifi_ssid,
+                                                  const char* wifi_password,
+                                                  const char* wifi_security_type );
+
+static int8_t
+validateXivelyCredentialBufferRequirement( const char* config_key, const char* value );
+
+/* CCS requirement */
+extern void ( *const g_pfnVectors[] )( void );
+
+/* Device and Simplelink WiFi Provisioning Functions */
+long MainLogic();
+void WaitForWlanEvent();
+static void BoardInit( void );
+static void InitializeAppVariables();
+static long ConfigureSimpleLinkToDefaultState();
+
+/****************************************************************************
+ *                               Source!
+ *****************************************************************************/
+
+/**
+ * @brief Main Application entry point.
+ *
+ * Initializes board and then calls MainLogic
+ * for Wifi Provisioning and connecting to Xively
+ */
+int main()
+{
+    long lRetVal = -1;
+
+    /* Initialize Board configurations */
+    BoardInit();
+    PinMuxConfig();
+
+    /* configure RED and Green LED */
+    GPIO_IF_LedConfigure( LED1 | LED2 | LED3 );
+
+#ifndef NOTERM
+    InitTerm();
+    ClearTerm();
+#endif
+
+    lRetVal = MainLogic();
+
+    if ( lRetVal < 0 )
+    {
+        ERR_PRINT( lRetVal );
+    }
+
+    LOOP_FOREVER();
+}
+
+/**
+ * @brief Driver for the xively library logic
+ *
+ * This function creates a xively context and starts the internal event
+ * dispatcher and event processing of Xively library.
+ *
+ */
+void ConnectToXively()
+{
+    /* Register Push Button Handlers */
+    Button_IF_Init( pushButtonInterruptHandlerSW2, pushButtonInterruptHandlerSW3 );
+
+    /* Make the buttons to react on both press and release events */
+    MAP_GPIOIntTypeSet( GPIOA1_BASE, GPIO_PIN_5, GPIO_BOTH_EDGES );
+    MAP_GPIOIntTypeSet( GPIOA2_BASE, GPIO_PIN_6, GPIO_BOTH_EDGES );
+
+    /* Disable the interrupt for now */
+    Button_IF_DisableInterrupt( SW2 | SW3 );
+
+    xi_state_t ret_state = xi_initialize( g_wifi_device_credentials.xivelyAccountId,
```

Comment:
after merging `development` branch into this branch the `xi_initialize` function will have one less argument. Please do the merge to make this branch "fresh".